File: tests/fixtures/lang/sample.erl
-module(sample).
-export([plain/0, if_else/1, case_demo/1, loops/1, bool_ops/3, try_catch/1, list_comp/1]).

plain() ->
    1.

if_else(X) ->
    if
        X > 0 -> 1;
        X < 0 -> 2;
        true   -> 3
    end.

case_demo(X) ->
    case X of
        1 -> 1;
        2 -> 2;
        _ -> 3
    end.

loops(N) ->
    S = lists:foldl(fun(I, Acc) -> Acc + I end, 0, lists:seq(0, N - 1)),
    if
        S > 10 -> S - 1;
        true   -> S
    end.

bool_ops(A, B, C) ->
    A andalso B orelse C.

try_catch(X) ->
    try
        case X of
            true  -> 1;
            false -> throw(error)
        end
    catch
        _:_ -> 3
    end.

list_comp(Xs) ->
    [X || X <- Xs, X > 0].

File: tests/fixtures/lang/sample.exs
defmodule Sample do

  def plain do
    1
  end

  def if_else do
    x = 0
    if x > 0 do
      1
    else
      if x < 0 do
        2
      else
        3
      end
    end
  end

  def cond_demo do
    x = 0
    cond do
      x > 0 -> 1
      x < 0 -> 2
      true  -> 3
    end
  end

  def case_demo do
    x = 0
    case x do
      1 -> 1
      2 -> 2
      _ -> 3
    end
  end

  def loops do
    s = 20
    if s > 10 do
      s - 1
    else
      s
    end
  end

  def bool_ops do
    a = true
    b = true
    c = false
    a && b || c
  end

  def try_catch do
    x = true
    try do
      case x do
        true  -> 1
        false -> throw(:error)
      end
    rescue
      _ -> 3
    end
  end

  def list_comp do
    xs = [1, 2, 3]
    for x <- xs, x > 0, do: x
  end

  def with_closure do
    x = 1
    add = fn a, b ->
      if a > b, do: a, else: b
    end
    add.(x, x)
  end

end
